src/lib/dateLib.ts

```diff
@@ -173,3 +178,206 @@ export const dateLib = {
   startOfWeek,
   startOfYear
 };
+
+/**
+ * The date library used by DayPicker. It's a subset of the date-fns functions
+ * with an explicit locale / formatting options.
+ *
+ * @private
+ * @internal
+ */
+export class DateLib {
+  /**
+   * Creates a new instance of the date library.
+   *
+   * @param options The formatting options for the date library.
+   * @param dateLibBaseOverrides Overrides for the date library functions.
+   */
+  constructor(
+    readonly options: Options = {},
+    dateLibBaseOverrides: Partial<DateLibBase> = {}
+  ) {
+    this.dateLibBase = {
+      ...defaultDateLibBase,
+      ...dateLibBaseOverrides
+    };
+  }
```

Comment:
Is the `dateLibBase` property necessary? What about a `DateLib` class to extends a `DefaultDateLib`: Then, in the constructor:

```ts
if ('addWeeks' in dateLibBaseOverrides) this.addWeeks = dateLibBaseOverrides.addWeeks;
```

There might be reasons why we’re not doing this, but I can’t identify any just by looking at the code.

Reply:
currently I've done it this way, as for `endOfWeek`, `format`, `getWeek`, and `startOfWeek` (locale required functions), I still want access to the original function with the correct name, but the interface from the class to not include the locale and have it pass in the internally known locale to the original function. I'll play around with another way to achieve the same thing a little cleaner.